pkgs/development/python-modules/agentops/default.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  setuptools,
+  fetchFromGitHub,
+  pythonOlder,
+  gitUpdater,
+
+  # dependencies
+  requests,
+  psutil,
+  packaging,
+  termcolor,
+  pyyaml,
+}:
+
+let
+  agentops = buildPythonPackage rec {
+    pname = "agentops";
+    version = "0.3.18";
+    pyproject = true;
+
+    disabled = pythonOlder "3.7";
+
+    src = fetchFromGitHub {
+      owner = "AgentOps-AI";
+      repo = "agentops";
+      rev = "refs/tags/${version}";
+      hash = "sha256-g3gKvCNGdeGyKCyTEG2Gss+lI5bGqkKftn8T4bB4c0c=";
+    };
+
+    passthru.updateScript = gitUpdater { };
+
+    pythonRelaxDeps = [
+      "psutil"
+      "packaging"
+    ];
+
+    enableParallelBuilding = true;
+
+    nativeBuildInputs = [ setuptools ];
+
+    dependencies = [
+      requests
+      psutil
+      packaging
+      termcolor
+      pyyaml
+    ];
+
+    doCheck = true;
+
+    pythonImportsCheck = [ "agentops" ];
+
+    meta = with lib; {
```

Comment:
We try to avoid `with` statements in front of attribute sets. Same remark for the rest of the files.

Reply:
I can change it, though it seems to be a very common idiom within the python packages at least.